dom.bs

```diff
@@ -1921,48 +1945,97 @@ is [=AbortSignal/aborted=]; otherwise false.
+<var>signals</var>, using <var>signalInterface</var>, which must be either {{AbortSignal}} or an
+interface that inherits from it, and an optional <var>realm</var>, run these steps:
 
 <ol>
- <li><p>If <var>followingSignal</var> is [=AbortSignal/aborted=], then return.
+ <li>
+  <p>Let <var>resultSignal</var> be a <a for=/>new</a> object implementing
+  <var>signalInterface</var> using <var>realm</var> if given, otherwise using the default behavior
+  defined in Web IDL.
 
- <li><p>If <var>parentSignal</var> is [=AbortSignal/aborted=], then
- <a for=AbortSignal>signal abort</a> on <var>followingSignal</var> with <var>parentSignal</var>'s
- [=AbortSignal/abort reason=].
+  <p class=XXX>As of the time of this writing Web IDL does not yet define any default behavior;
+  see <a href="https://github.com/whatwg/webidl/issues/135">whatwg/webidl#135</a>.
+
+ <li><p>Set <var>resultSignal</var>'s [=AbortSignal/composite=] to true.
+
+ <li><p><a for=list>For each</a> <var>signal</var> of <var>signals</var>: if <var>signal</var> is
+ [=AbortSignal/aborted=], then set <var>resultSignal</var>'s [=AbortSignal/abort reason=] to
+ <var>signal</var>'s [=AbortSignal/abort reason=] and return <var>resultSignal</var>.
 
  <li>
-  <p>Otherwise, <a for=AbortSignal lt=add>add the following abort steps</a> to
-  <var>parentSignal</var>:
+  <p><a for=list>For each</a> <var>signal</var> of <var>signals</var>:
 
   <ol>
-   <li><p><a for=AbortSignal>Signal abort</a> on <var>followingSignal</var> with
-   <var>parentSignal</var>'s [=AbortSignal/abort reason=].
+   <li>
+    <p>If <var>signal</var>'s [=AbortSignal/composite=] is false, then:
+
+    <ol>
+     <li><p><a for=set>Append</a> <var>signal</var> to <var>resultSignal</var>'s
+     [=AbortSignal/source signals=].
+
+     <li><p><a for=set>Append</a> <var>resultSignal</var> to <var>signal</var>'s
+     [=AbortSignal/dependent signals=].
+    </ol>
+
+   <li>
+    <p>Otherwise, <a for=list>for each</a> <var>sourceSignal</var> of <var>signal</var>'s
+    [=AbortSignal/source signals=]:
+
+    <ol>
+     <li><p>Assert: <var>sourceSignal</var> is not [=AbortSignal/aborted=] and not
+     [=AbortSignal/composite=].
+
+     <li><p>If <var>resultSignal</var>'s [=AbortSignal/source signals=] <a for=set>contains</a>
+     <var>sourceSignal</var>, then <a for=iteration>continue</a>.
+
+     <li><p><a for=set>Append</a> <var>sourceSignal</var> to <var>resultSignal</var>'s
+     [=AbortSignal/source signals=].
+
+     <li><p><a for=set>Append</a> <var>resultSignal</var> to <var>sourceSignal</var>'s
+     [=AbortSignal/dependent signals=].
+    </ol>
   </ol>
+ <li>Return <var>resultSignal</var>.
 </ol>
 
 
+<h4 id=abort-signal-garbage-collection>Garbage collection</h4>
+
+<p>A [=AbortSignal/composite=] {{AbortSignal}} object must not be garbage collected while its
+[=AbortSignal/source signals=] is non-empty unless it has no registered event listeners for its
```

Comment:
Maybe "and it has registered ... or its abort algorithms is non-empty"?

Reply:
Yeah that's probably better. I think "unless ..." was me trying to make the "and ... or" unambiguous, but it's a bit awkward.